pallets/parachain-staking/src/lib.rs

```diff
@@ -1448,6 +1479,34 @@ pub mod pallet {
 			});
 			Ok(().into())
 		}
+		/// Batch migrate locks to freezes for a list of accounts.
+		///
+		/// This function allows migrating multiple accounts from the old lock-based
+		/// staking to the new freeze-based staking in a single transaction.
+		///
+		/// Parameters:
+		/// - `accounts`: List of tuples containing (account_id, is_collator)
+		///   where is_collator indicates if the account is a collator (true) or delegator (false)
+		///
+		/// The maximum number of accounts that can be migrated in one batch is 100.
+		#[pallet::call_index(33)]
+		#[pallet::weight({			
+			const MAX_ACCOUNTS: u32 = 100;
+			T::WeightInfo::migrate_locks_to_freezes_batch_delegators(MAX_ACCOUNTS).max(T::WeightInfo::migrate_locks_to_freezes_batch_candidates(MAX_ACCOUNTS))
+		})]
+		pub fn migrate_locks_to_freezes_batch(
+			origin: OriginFor<T>,
+			accounts: BoundedVec<(T::AccountId, bool), ConstU32<100>>,
```

Comment:
Maybe use the constant everywhere and give it a more precise name
```suggestion
			accounts: BoundedVec<(T::AccountId, bool), ConstU32<MAX_ACCOUNTS_PER_MIGRATION_BATCH>>,
```

Reply:
Amended https://github.com/moonbeam-foundation/moonbeam/pull/3306/commits/df2ebaa7897f875c07efd8bf01133e5d5f1cb2bb